core/src/test/scala/scalan/TestContexts.scala

```diff
@@ -43,7 +43,7 @@ trait TestContexts extends TestsUtil {
     import compiler._
 
     def test[A,B](functionName: String, f: => scalan.Exp[A => B]): CompilerOutput[A, B] = {
-      buildExecutable(FileUtil.file(prefix, functionName), functionName, f, GraphVizConfig.default)(defaultCompilerConfig)
+      buildExecutable(FileUtil.file(prefix + "/" + testName, functionName), functionName, f, GraphVizConfig.default)(defaultCompilerConfig)
```

Comment:
This will have the same effect. Actually, why not `val file = if (testName == functionName) FileUtil.file(prefix, functionName) else FileUtil.file(prefix, testName, functionName)`? No need to create extra dir to navigate to in the common case.